kratos/solving_strategies/schemes/residual_based_adjoint_bossak_scheme.h

```diff
@@ -222,23 +286,55 @@ class ResidualBasedAdjointBossakScheme : public Scheme<TSparseSpace, TDenseSpace
         KRATOS_CATCH("");
     }
 
-    void CalculateSystemContributions(Condition& rCurrentCondition,
-                                      LocalSystemMatrixType& rLHS_Contribution,
-                                      LocalSystemVectorType& rRHS_Contribution,
-                                      Condition::EquationIdVectorType& rEquationId,
-                                      const ProcessInfo& rCurrentProcessInfo) override
+    void CalculateSystemContributions(
+        Condition& rCurrentCondition,
+        LocalSystemMatrixType& rLHS_Contribution,
+        LocalSystemVectorType& rRHS_Contribution,
+        Condition::EquationIdVectorType& rEquationId,
+        const ProcessInfo& rCurrentProcessInfo) override
     {
         KRATOS_TRY;
-        // NOT TESTED !!!
-        rCurrentCondition.CalculateLocalSystem(
-            rLHS_Contribution, rRHS_Contribution, rCurrentProcessInfo);
+
+        const auto k = OpenMPUtils::ThisThread();
+        rCurrentCondition.GetValuesVector(mAdjointValuesVector[k]);
+        const auto local_size = mAdjointValuesVector[k].size();
+        if (rRHS_Contribution.size() != local_size)
+        {
+            rRHS_Contribution.resize(local_size, false);
+        }
+        if (rLHS_Contribution.size1() != local_size || rLHS_Contribution.size2() != local_size)
+        {
+            rLHS_Contribution.resize(local_size, local_size, false);
+        }
+        this->CheckAndResizeThreadStorage(local_size);
+
+        this->CalculateGradientContributions(rCurrentCondition, rLHS_Contribution,
+                                             rRHS_Contribution, rCurrentProcessInfo);
+
+        this->CalculateFirstDerivativeContributions(
+            rCurrentCondition, rLHS_Contribution, rRHS_Contribution, rCurrentProcessInfo);
+
+        this->CalculateSecondDerivativeContributions(
+            rCurrentCondition, rLHS_Contribution, rRHS_Contribution, rCurrentProcessInfo);
+
+        // It is not required to call CalculatePreviousTimeStepContributions here again
+        // since, the previous time step contributions from conditions are stored in variables
+        // mentioned in AdjointExtensions, and they are added via CalculateSystemContributions<ElementType>
+        // method.
```

Comment:
what happens if I didn't call the corresponding element methods before?
Is this always going to be the case?

Reply:
In the case of Fluids, I would say, we always call the elements one. But I am not sure about the structural adjoints cases, `Condition`s were anyway not supported untill now. @MFusseder may be we can agree or change this?

Reply:
in structural we also always call all elements.